purchase_rfq_number/models/purchase_order.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 ProThai Technology Co.,Ltd. (http://prothaitechnology.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+import base64
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    rfq_number = fields.Char(
+        "RFQ Reference",
+        index=True,
+        copy=False,
+        default="New",
+    )
+
+    @api.model
+    def create(self, vals):
+        company = False
+        if "company_id" in vals:
+            company = self.env["res.company"].browse(vals.get("company_id"))
+        else:
+            company = self.env.company
+        if not company.keep_name_po and vals.get("name", "New") == "New":
+            vals["name"] = self.env["ir.sequence"].next_by_code("purchase.rfq") or "New"
+            vals["rfq_number"] = vals["name"]
+        return super().create(vals)
+
+    def button_confirm(self):
```

Comment:
Can you shorten the code (make it work as multi).
**Note:** I saw this code in sale_quotation_number, but I think we can make it better.